SparkPost/matchbox
# FE-792 Add Stack layout component

### What Changed
- Adds a new layout component, Stack
- [See ticket for details](https://jira.int.messagesystems.com/browse/FE-767)

### Props: 
**Space**
- Should accept either a string or a responsive array (See styled-system responsive props)
- Should accept the same set of 'space' tokens defined in the styled system theme
  - 0, 100, 200, 300, 400, 500, 600, 700, 800

**Align**
- Should accept either a string or a responsive array (See styled-system responsive props) 
- Should set the horizontal alignment of Stack's direct children
  - center, left, right

### How To Test or Verify
- Run Storybook and visit the Stack stories, starting with: http://localhost:9001/?path=/story/layout-stack--spacing
- Verify the responsive props behave as expected

<img width="1003" alt="Screen Shot 2020-01-03 at 9 50 19 AM" src="https://user-images.githubusercontent.com/3903325/71729777-93a97e80-2e0e-11ea-9346-a1c8955df771.png">


### PR Checklist
